src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -16,16 +16,40 @@
 
 namespace hyrise {
 
-void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start) {
+
+void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start,
+                          std::atomic<bool>& threads_ready_to_be_executed) {
+  while (!threads_ready_to_be_executed) {}
   const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
-  const auto bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
-  lseek(fd, from * uint32_t_size, SEEK_SET);
-  Assert((read(fd, read_data_start + from, bytes_to_read) == bytes_to_read),
-         close_file_and_return_error_message(fd, "Read error: ", errno));
+  const auto total_bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
+  const auto elements_to_read = static_cast<uint64_t>(total_bytes_to_read / uint32_t_size);
+  const auto MAX_NUMBER_OF_ELEMENTS = uint64_t{536'869'888};
+
+  if (elements_to_read > MAX_NUMBER_OF_ELEMENTS) {
+    auto elements_read = uint64_t{0};
```

Comment:
You can calculate elements_read out of elements_to_read and elements_remaining.
But I am not sure if this is easier to read.